synapseclient/retry.py

```diff
@@ -46,7 +53,7 @@ def foo(a, b, c): return [a, b, c]
             response = function()
         except Exception as ex:
             exc_info = sys.exc_info()
-            log_error(str(ex), verbose)
+            logger.debug("calling %s resulted in an Exception" % function.__name__)
```

Comment:
this is failing a unit test because of magicmock. add a __name__ to the magicmock object